### ISSUE TITLE
Better errors when accidentally calling type-assertion expressions

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#26991]
```ts
declare function foo(): string;

foo()
(document.getElementById("lelement") as HTMLDivElement).id;
```

@mousetraps just ran into this totally garbage error message

```
[ts] Cannot invoke an expression whose type lacks a call signature. Type 'String' has no compatible call signatures.
```

Given

1. A call expression with one argument
1. whose open paren starts on the next line
1. and whose sole argument is an `AsExpression`/`TypeAssertionExpression`

we should give a specialized error that ends in:

```
It is highly likely that you are missing a semicolon.
```

**Comment by Kingwl** [comment]
should that work with all ASI context?

**Comment by DanielRosenwasser** [comment]
It's sort of tough to nail down because not all of these are type errors off the bat. The other one that comes to mind is accessing a member off of an array which *actually* got parsed as an element access with a comma expression:

```ts
foo()
[1, 2, 3].forEach(x => {
  return x * x;
})
```

Related: https://twitter.com/drosenwasser/status/939589182833410048

**Comment by DanielRosenwasser** [comment]
I also want to say that this improvement is definitely higher-priority. If a user weren't using TypeScript, then it's significantly less likely that they'd have run into this problem.

**Comment by DanielRosenwasser** [comment]
Thanks a bunch @Kingwl! 💯
